mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt

```diff
@@ -29,6 +29,55 @@ inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.fragm
         .apply { subscribe(this@fragmentViewModel, subscriber = { postInvalidate() }) }
 }
 
+/**
+ * Gets or creates a ViewModel scoped to a parent fragment. This delegate will walk up the parentFragment hierarchy
+ * until it finds a Fragment that can provide the correct ViewModel. If no parent fragments can provide the ViewModel,
+ * a new one will be created in top-most parent Fragment.
+ */
+inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.parentFragmentViewModel(
+    viewModelClass: KClass<VM> = VM::class,
+    crossinline keyFactory: () -> String = { viewModelClass.java.name }
+): Lazy<VM> where T : Fragment, T : MvRxView = lifecycleAwareLazy(this) {
+    requireNotNull(parentFragment) { "There is no parent fragment!" }
+    val factory = MvRxFactory { error("No ViewModel for this Fragment.") }
+    var fragment: Fragment? = parentFragment
+    val key = keyFactory()
+    while (fragment != null) {
+        try {
+            return@lifecycleAwareLazy ViewModelProviders.of(fragment, factory).get(key, viewModelClass.java)
+                .apply { subscribe(this@parentFragmentViewModel, subscriber = { postInvalidate() }) }
+        } catch (e: IllegalStateException) {
+            fragment = fragment.parentFragment
+        }
+    }
+
+    // ViewModel was not found. Create a new one in the top-most parent.
+    var topParentFragment = parentFragment
+    while (topParentFragment?.parentFragment != null) {
+        topParentFragment = topParentFragment.parentFragment
+    }
+    val viewModelContext = FragmentViewModelContext(this.requireActivity(), _fragmentArgsProvider(), topParentFragment!!)
+    return@lifecycleAwareLazy MvRxViewModelProvider.get(viewModelClass.java, S::class.java, viewModelContext, keyFactory())
+        .apply { subscribe(this@parentFragmentViewModel, subscriber = { postInvalidate() }) }
+}
+
+/**
+ * Gets or creates a ViewModel scoped to a target fragment. Throws [IllegalStateException] if there is no target fragment.
```

Comment:
this use case seems weird to me, because the lifecycle of the target fragment isn't necessarily tied to the fragment that is calling this. Couldn't the target fragment be destroyed, and the viewmodel cleared, while this fragment is still using it?

is it that bad to share viewmodels via an activity scoped viewmodel in this case?

I'm not particularly opposed to it, but am curious to hear more about the need. Once we have these new delegates it would be great if we updated the wiki with a little guide about when we recommend using each

Reply:
@elihart I think it alleviates some of the potential ambiguity in the parentFragment case because it allows you to be explicit what the ViewModel will be scoped to.

I haven't actually needed it myself but @littleGnAl suggested it and it seems reasonable and was simple enough.

Reply:
@gpeal @elihart In our project, the `targetFragment` is generally used with `DialogFragment`, but MvRx does not officially support `DialogFragment`, I think this is not necessary.

 If someone needs to get the `ViewModel` of the `targetFragment `, they can explicitly cast the `targetFragment` such as `(targetFragment as MyTargetFragment).myTargetViewModel`. 

But if this function is needed, I think the logic should be similar to the logic of `parentFragmentViewModel`, because the `targetFragment`'s `ViewModel` maybe already exist.

Reply:
@littleGnAI, ah, the DialogFragment case makes sense to me